academy/xl-run-prod/2-keys.md

```diff
@@ -0,0 +1,76 @@
+---
+title: Prepare a validator and keys
+order: 2
+description: Set up keys for use for your validator
+tag: deep-dive
+---
+
+# Prepare a validator and keys
+
+In the [previous section](./1-software.md), you prepared a binary for your nodes. Some of your nodes will be validators. To propose and sign blocks, validators need access to private keys on an on-going basis. A regular node does not need such keys.
+
+Here you learn how to prepare a validator and handle its keys. This works whether you are preparing a validator to join a preexisting network, or you are setting up your validator to be part of the [genesis](./genesis.md).
+
+## Private key security considerations
+
+More precisely than needing access to private keys, validators only need the capability to sign blocks on an on-going basis. There is a security difference between access to the private key and access to a signing facility:
+
+1. When your validator has access to the private key, if your validator node has been compromised, your private key is too, and you are at the risk of wrongfully signing malicious blocks **for ever**.
+2. On the other hand, when you only provide a signing _service_ to your validator, and your validator node has been compromised, you are _only_ at the risk of wrongfully signing malicious blocks **for as long as the signing service is up**.
+
+In order to mitigate **point 1**, you can keep your private key in a [hardware security module](https://hub.cosmos.network/main/validators/validator-faq.html#how-to-handle-key-management) (a.k.a. HSM), from which it can be retrieved only once, during the HSM's offline setup. This HSM device then remains plugged into the computer that runs the validator or the signing service. See [here](https://hub.cosmos.network/main/validators/security.html#key-management-hsm) for the current list of supported devices. To use an HSM you own, you need physical access to the computer into which you plug it in.
+
+In order to mitigate **point 2**, you can use a specialized [key management system](https://hub.cosmos.network/main/validators/kms/kms.html) (a.k.a. KMS). It runs on a computer separate from your validator node, it has access to the hardware key and is [contacted over the private network](https://github.com/iqlusioninc/tmkms/blob/v0.12.2/README.txsigner.md#architecture) by your validator node(s) for the purpose of signing blocks. Such a KMS is specialized in the sense that it is, for instance, able to detect attempts are signing two different blocks at the same height.
+
+You can combine these stragegies. For instance, if you insist on using an HSM and having your validator node located in the cloud, you can run the KMS on a computer to which you have physical access, and which dials into your remote validator node to provide the signing service.
```

Comment:
And, is "the computer to which you have physical access" that runs the KMS **_also_** the computer into which the HSM is plugged?